lib/puppet/agent.rb

```diff
@@ -117,7 +119,7 @@ def with_client(transaction_uuid, job_id = nil)
       @client = client_class.new(transaction_uuid, job_id)
     rescue StandardError => detail
       Puppet.log_exception(detail, _("Could not create instance of %{client_class}: %{detail}") % { client_class: client_class, detail: detail })
-      return
+      nil
```

Comment:
I think we should keep this `return` as https://github.com/puppetlabs/puppet/pull/8125/files#diff-ac44b8f38a4592c950df5cdc74689d93L122 should not execute and this function return value is never used

Reply:
Yes, i also thought about that, seems better that way